openebs/Mayastor
# feat: configure for direnv via envrc

This PR enables [`direnv`](https://direnv.net/) users to, when changing into the directory:

* (For the first time) See that we have an `.envrc` file. They can then type `direnv allow .` to give permission to their system to autoload the `.envrc`.
* Have `direnv`, a tool which has a shell hook, load the requisite `shell.nix` automatically.

This is beneficial to users who use `direnv` (which I understand is common in the nix world) as once a user has configured this, they no longer need to type `nix-shell` to activate their environment. Some users also use IDE tools like [VSCode direnv](https://marketplace.visualstudio.com/items?itemName=Rubymaniac.vscode-direnv) for this.

# To test

Try adding `direnv` to your package set ([don't forget to add the hook to your shell]( https://direnv.net/docs/hook.html)).

Then, `cd` into your Mayastor directory from a different one. It might do some extra work, but your output should look like this:

```bash
[ana@architect:~/git/openebs]$ cd Mayastor/
direnv: loading ~/git/openebs/Mayastor/.envrc
direnv: using nix
direnv: using cached derivation
direnv: eval /home/ana/git/openebs/Mayastor/.direnv/cache-pre262347.257cbbcd3ab
direnv: export +AR +AR_FOR_TARGET +AS +AS_FOR_TARGET +CC +CC_FOR_TARGET +CONFIG_SHELL +CXX +CXX_FOR_TARGET +DETERMINISTIC_BUILD +HOST_PATH +IN_NIX_SHELL +LD +LD_FOR_TARGET +LIBCLANG_PATH +NIX_BINTOOLS +NIX_BINTOOLS_FOR_TARGET +NIX_BINTOOLS_WRAPPER_TARGET_HOST_x86_64_unknown_linux_gnu +NIX_BINTOOLS_WRAPPER_TARGET_TARGET_x86_64_unknown_linux_gnu +NIX_BUILD_CORES +NIX_BUILD_TOP +NIX_CC +NIX_CC_FOR_TARGET +NIX_CC_WRAPPER_TARGET_HOST_x86_64_unknown_linux_gnu +NIX_CC_WRAPPER_TARGET_TARGET_x86_64_unknown_linux_gnu +NIX_CFLAGS_COMPILE +NIX_CFLAGS_COMPILE_FOR_TARGET +NIX_ENFORCE_NO_NATIVE +NIX_HARDENING_ENABLE +NIX_INDENT_MAKE +NIX_LDFLAGS +NIX_LDFLAGS_FOR_TARGET +NIX_PKG_CONFIG_WRAPPER_TARGET_TARGET_x86_64_unknown_linux_gnu +NIX_STORE +NM +NM_FOR_TARGET +NODE_PATH +OBJCOPY +OBJCOPY_FOR_TARGET +OBJDUMP +OBJDUMP_FOR_TARGET +PKG_CONFIG_FOR_TARGET +PKG_CONFIG_PATH_FOR_TARGET +PROTOC +PROTOC_INCLUDE +PYTHONHASHSEED +PYTHONNOUSERSITE +PYTHONPATH +RANLIB +RANLIB_FOR_TARGET +READELF +READELF_FOR_TARGET +SIZE +SIZE_FOR_TARGET +SOURCE_DATE_EPOCH +SPDK_PATH +STRINGS +STRINGS_FOR_TARGET +STRIP +STRIP_FOR_TARGET +TEMP +TMP +TMPDIR +__ETC_PROFILE_SOURCED +buildInputs +builder +configureFlags +depsBuildBuild +depsBuildBuildPropagated +depsBuildTarget +depsBuildTargetPropagated +depsHostHost +depsHostHostPropagated +depsTargetTarget +depsTargetTargetPropagated +doCheck +doInstallCheck +hardeningDisable +name +nativeBuildInputs +nobuildPhase +out +outputs +patches +phases +propagatedBuildInputs +propagatedNativeBuildInputs +shell +shellHook +stdenv +strictDeps +system ~PATH

[ana@architect:~/git/openebs/Mayastor]$ cd ..
direnv: unloading
```

Now you should be able to skip running `nix-shell` to enter your dev environment.
